modules/dnn/src/layers/layers_common.simd.hpp

```diff
@@ -121,7 +122,66 @@ void fastConv( const float* weights, size_t wstep, const float* bias,
                    vs20 = _mm256_setzero_ps(), vs21 = _mm256_setzero_ps(),
                    vs22 = _mm256_setzero_ps(), vs23 = _mm256_setzero_ps();
 
-            for( int k = 0; k < vecsize; k += 8, rptr += 8 )
+#if CV_AVX512_SKX // AVX512VL is necessary to avoid register spilling
+           /* only use AVX512 for multiple-of-16 vectors */
+           if ((vecsize & 15) == 0) {
+
+                    __m512 vs00_5 = _mm512_setzero_ps(), vs01_5 = _mm512_setzero_ps(),
+                           vs02_5 = _mm512_setzero_ps(), vs03_5 = _mm512_setzero_ps(),
+                           vs10_5 = _mm512_setzero_ps(), vs11_5 = _mm512_setzero_ps(),
+                           vs12_5 = _mm512_setzero_ps(), vs13_5 = _mm512_setzero_ps(),
+                           vs20_5 = _mm512_setzero_ps(), vs21_5 = _mm512_setzero_ps(),
+                           vs22_5 = _mm512_setzero_ps(), vs23_5 = _mm512_setzero_ps();
+
+                    for(; k < vecsize - 15; k += 16, rptr += 16 )
+                    {
+                        __m512 w0 = _mm512_load_ps(wptr0 + k);
+                        __m512 w1 = _mm512_load_ps(wptr1 + k);
+                        __m512 w2 = _mm512_load_ps(wptr2 + k);
+                        __m512 r0 = _mm512_load_ps(rptr);
```

Comment:
`_mm512_load_ps()` is aligned load. Need to check that ptrs are aligned to 64-bytes or to change this to `_mm512_loadu_ps()`.
Currently these ptrs are 32-bytes aligned only (based on OpenCV's memory allocator alignment requirement), so AVX/AVX2 code is fine here.